src/DataGrid.tsx

```diff
@@ -393,16 +651,34 @@ function DataGrid<R, K extends keyof R, SR>({
           row={row}
           viewportColumns={viewportColumns}
           lastFrozenColumnIndex={lastFrozenColumnIndex}
+          selectedCellIdx={selectedPosition.rowIdx === rowIdx && selectedPosition.mode === 'SELECT' ? selectedPosition.idx : undefined}
+          copiedCellIdx={copiedPosition?.rowIdx === rowIdx ? copiedPosition.idx : undefined}
+          draggedOverCellIdx={isDraggedOver(rowIdx) ? selectedPosition.idx : undefined}
           eventBus={eventBus}
           isRowSelected={isRowSelected}
           onRowClick={onRowClick}
           rowClass={rowClass}
           top={rowIdx * rowHeight + totalHeaderHeight}
+          enableCellDragAndDrop={enableCellDragAndDrop}
         />
       );
     }
 
-    return rowElements;
+    return (
+      <div
+        onKeyDown={handleKeyDown}
+      >
+        {rowElements}
+        {getEditorContainer()}
+      </div>
+    );
+  }
+
+  // Reset the positions if the current values are no longer valid. This can happen if a column or row is removed
+  if (selectedPosition.idx > columns.length || selectedPosition.rowIdx > rows.length) {
+    setSelectedPosition({ idx: -1, rowIdx: -1, mode: 'SELECT' });
```

Comment:
I'd say we create this `{ idx: -1, rowIdx: -1, mode: 'SELECT' }` to a separate standalone variable to be reused.

https://github.com/adazzle/react-data-grid/pull/2061/files#diff-60532bd8286dc6c925b8ea6ee3e2573bR241